lib/blob.js

```diff
@@ -0,0 +1,377 @@
+ * @param {object} header - Mapping from header key in lowercase to value.
+ * @return {Promise} A promise for HTTPS response with `payload` property as
+ * string containing the response payload.
+ */
+Blob.prototype.request = function request(method, path, query, headers, data) {
+  // Set timeout, if not provided
+  if (query.timeout === undefined) {
+    query.timeout = this.timeoutInSeconds;
+  }
+
+  // Set date, version and client-request-id headers
+  headers['x-ms-date']              = new Date().toUTCString();
+  headers['x-ms-version']           = this.options.version;
+  headers['x-ms-client-request-id'] = this.options.clientId;
+
+  // Construct authorized request options with shared key signature or
+  // shared-access-signature.
+  var self = this;
+  return this.authorize(method, path, query, headers).then(function(options) {
+    // Retry with retry policy
+    return utils.retry(function(retry) {
+      debug("Request: %s %s, retry: %s", method, path, retry);
+
+      // Construct a promise chain first handling the request, and then parsing
+      // any potential error message
+      return utils.request(options, data, self.timeout).then(function(res) {
+        // Accept the response if it's 2xx, otherwise we construct and
+        // throw an error
+        if (200 <= res.statusCode && res.statusCode < 300) {
+          return res;
+        }
+
+        // Parse error message
+        // TODO rename the function to something generic
+        var data = xml.queueParseError(res);
+
+        // Construct error object
+        var err         = new Error(data.message);
+        err.name        = data.code + 'Error';
+        err.code        = data.code;
+        err.statusCode  = res.statusCode;
+        err.message     = data.message;
+        err.retries     = retry;
+
+        debug("Error code: %s for %s %s on retry: %s",
+          err.code, method, path, retry);
+
+        // Throw the constructed error
+        throw err;
+      });
+    }, self.options);
+  });
+};
+
+/**
+ * Create a new container with the given 'name' under the storage account.
+ *
+ * @method createContainer
+ * @param {string} name -  Name of the container to create
+ * @param {object} metadata - Mapping from metadata keys to values.
+ * @param {string} publicLevelAccess - Specifies whether data in the container may be accessed publicly and the level of access.
+ *                                    Possible values: container, blob.
+ * @returns {Promise} A promise that container has been created.
+ */
+Blob.prototype.createContainer = function createContainer(name, metadata, publicLevelAccess) {
+  // Construct headers
+  var headers = {};
+  for(var key in metadata) {
+    if (metadata.hasOwnProperty(key)) {
+      headers['x-ms-meta-' + key] = metadata[key];
+    }
+  }
+
+  if(publicLevelAccess && publicLevelAccess === 'container' || publicLevelAccess === 'blob') {
+    headers['x-ms-blob-public-access'] = publicLevelAccess;
+  }
```

Comment:
You are right! I will throw an error if the user sets an incorrect level of access.